workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -24,7 +24,32 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
+            // Represents a bit mask for tracking which component properties were changed locally and need to be synced.
+            [Flags]
+            public enum DirtyBitsFlag
+            {
+                None = 0,
+<# for(int i = 0; i < fieldDetailsList.Count; i++) { #>
+                <#= fieldDetailsList[i].PascalCaseName #> = <#= 1 << i #>,
```

Comment:
I think we can do a field count check and either use an int32 or in64 enum? (the 2 supported options in C#)
Any component with more then 64 fields should fail to generate. I doubt limiting Components to only 64 fields would be a painful limitation?

Reply:
I've asked production (https://improbable.slack.com/archives/C9PR5EA21/p1540377195000100) and it looks like nobody right now is using more than 64 fields. There are hypothetical use cases where you define elements of a fixed length collection as explicit fields to avoid using schema lists/collections but that use case should have a limited shelf life with delta compressed component updates around the corner. My suggestion would be: Let's go with @zeroZshadow's suggestion for now, add a known issue note in the docs and see how many people actually complain about this limitation. This might end up being a known issue that we'll never fix. @samiwh  are you on-board with this?

Reply:
Sounds fair. I will look into stackallocing a byte array.

Reply:
stackalloc is not something that works here.
If we want beyond 64 fields, we'd need have to have multiple bit flag fields.
I still think a limitation of 64 fields is a totally valid restraint